chapters/chap03-sec02.tex

```diff
@@ -18,17 +18,97 @@ \subsection{\mimc{} Encryption}\label{instantiation:mkhash:mimc-encryption}
+
+In fact, when considering security against algebraic attacks, we want (roughly speaking) the polynomial that defines the cipher to be of maximum degree and full (or at least, very dense). That is we want the degree of the polynomial to be higher than $2^{\secpar}$. Since in $\mimc{}$ the S-box is defined as $S(x) = x^{\exponent{}}$, then after $\rounds$ rounds the degree of the polynomial describing the cipher is $\exponent{}^{\rounds}$. Hence, we need
+
+\[
+    \exponent{}^{\rounds} \geq 2^{\secpar}
+\]
+that is, we want
+\[
+    \rounds \geq \secpar \log_{\exponent}(2)
+\]
+
+\begin{remark}
+    More rounds may be required as advised in~\cite{DBLP:conf/asiacrypt/Eichlseder0LORS20} to prevent some algebraic attacks that can be mounted when $\mimc{}$ is used over binary fields $\FFx{2^n}$.
+\end{remark}
+
+It is important to note that the security analysis related to algebraic attacks relies on the fact that the polynomial describing the cipher is dense/full. If this assumption is violated, a more granular security analysis needs to be carried for the setting of interest.
+
+\begin{notebox}
+    For \emph{small exponents and large prime fields} (e.g.~for $\secpar = 128$, $p = 2^{128}$ and $\exponent{} = 3$), we see that the lower bound on the number of rounds is (much) smaller in the context of statistical attacks than in the context of algebraic attacks. As such, we see that in such settings algebraic attacks are much more powerful than statistical attacks. Hence, when instanting $\mimc{}$ with an \emph{small} exponent of the form $2^t \pm 1$, it is crucial to make sure that, even if the resulting map is not an APN, the polynomial describing the cipher remains full/very dense. Importantly, if the setting is changed (e.g.~to use exponents that are ``big'' w.r.t.~the field size) the security analysis proposed in~\cite{albrecht2016mimc} must be changed.
+\end{notebox}
+
+\medskip
+
+In the sections below, and as in ~\cite{albrecht2016mimc}, we will consider exponents of the form $\exponent{} = 2^t-1$ and $\exponent{} = 2^t+1$ where $\gcd(\exponent{}, \rCURVE-1) = 1$ (such that exponentiation by $\exponent{}$ yields a permutation on $\FFx{\rCURVE}$). We note that the term cancellation happening with exponents of the form $\exponent{} = 2^t+1$ does not immediatelly translate to the context where $\mimc{}$ is carried out over prime fields of large odd characteristic. In fact, in the case of $\FFx{\rCURVE}$, where $\rCURVE > \binom{\exponent}{\lfloor \exponent / 2
```

Comment:
```suggestion
In the sections below, and as in ~\cite{albrecht2016mimc}, we will consider exponents of the form $\exponent{} = 2^t-1$ and $\exponent{} = 2^t+1$ where $\gcd(\exponent{}, \rCURVE-1) = 1$ . We note that the term cancellation happening with exponents of the form $\exponent{} = 2^t+1$ does not immediatelly translate to the context where $\mimc{}$ is carried out over prime fields of large odd characteristic. In fact, in the case of $\FFx{\rCURVE}$, where $\rCURVE > \binom{\exponent}{\lfloor \exponent / 2
```
Removed "(such that exponentiation by $\exponent{}$ yields a permutation on $\FFx{\rCURVE}$)" asit was alraedy explaned why we need gcd to be 1